src/stream/src/executor/project.rs

```diff
@@ -96,11 +96,11 @@ impl Inner {
         &self,
         chunk: StreamChunk,
     ) -> StreamExecutorResult<Option<StreamChunk>> {
-        let chunk = if chunk.selectivity() <= self.materialize_selectivity_threshold {
-            chunk.compact()
-        } else {
-            chunk
-        };
+        // let chunk = if chunk.selectivity() <= self.materialize_selectivity_threshold {
```

Comment:
This is somewhat a workaround before. As we now compact all data after `Exchange`, the problem should be mostly resolved.